package/com.unity.formats.usd/Tests/Editor/InitUsdTests.cs

```diff
@@ -0,0 +1,61 @@
+using System.IO;
+using NUnit.Framework;
+using System.Reflection;
+using Unity.Formats.USD;
+using UnityEngine;
+
+public class InitUsdTests
+{
+    [Test]
+    [Ignore("[USDU-249]")]
+    public void SetupUsdPath_InvalidPath_Error()
+    {
+        var invalidFilePath = "\\NonExisting\\Path\\";
+
+        // SetupUsdPath function is a private function
+        var setUpMethod = GetMethod("SetupUsdPath");
+
+        var expectedError = Assert.Throws<TargetInvocationException>(() => setUpMethod.Invoke(null, new object[] { invalidFilePath }), "Error was expected but not thrown");
+        Assert.IsInstanceOf<FileNotFoundException>(expectedError.InnerException);
+        Assert.AreEqual(string.Format("Could not find file '{0}'.", invalidFilePath), expectedError.InnerException.Message, "Unexpected error message was given");
+    }
+
+    [Test]
+    public void SetupUsdPath_EmptyPath_Error()
+    {
+        // SetupUsdPath function is a private function
+        var setUpMethod = GetMethod("SetupUsdPath");
+
+        var expectedError = Assert.Throws<TargetInvocationException>(() => setUpMethod.Invoke(null, new object[] { "" }), "Error was expected but not thrown");
+        Assert.IsInstanceOf<System.ArgumentException>(expectedError.InnerException);
+        Assert.AreEqual("The specified path is not of a legal form (empty).", expectedError.InnerException.Message, "Unexpected error message was given");
+    }
+
+    [Test]
+    public void InitUsd_Initialize()
+    {
+        // Reset 'm_usdInitialized' for accurate testing
+        ResetInitUsd();
+
+        Assert.True(InitUsd.Initialize(), "USD Initialize failed");
```

Comment:
Is there a more robust thing we can also check for, as well as the return value of Initialise()?

Reply:
eg check that the plugin is registered, that the expected debug log is printed, that no other errors are handled and logged.

Reply:
i've looked into this a bit more - on how to tell if the plugin has been registered and the typebinding
But I just ended up having more questions, do you know who I should be asking about the init steps?

Reply:
You can check that unity types have been added the TypeBinder, that usd error message are logged in Unity console and that the plugins Registry is not empty. We can chat if you have more questions.

Reply:
After talking with Julien, it turns out the PlugRegistry is not behaving correctly - created a separate JIRA ticket for this
https://jira.unity3d.com/browse/USDU-295